### PR TITLE
VPN revisions

### PR DESCRIPTION
**Pull request by unwin**

### CODE REVIEW

File: src/main/java/org/dasein/cloud/network/AbstractVPNSupport.java [hunk 1]
```diff
@@ -31,11 +31,12 @@ public VPNGateway connectToVPNGateway(String vpnName, String endpoint, String na
     }
 
     @Deprecated
-    public @Nonnull VPN createVPN(@Nullable String inProviderDataCenterId, @Nonnull String name, @Nonnull String description, @Nonnull VPNProtocol protocol) throws CloudException, InternalException {
-        throw new OperationNotSupportedException("createVPN is not currently implemented in " + getProvider().getCloudName());
+    public final @Nonnull VPN createVPN(@Nullable String inProviderDataCenterId, @Nonnull String name, @Nonnull String description, @Nonnull VPNProtocol protocol) throws CloudException, InternalException {
+        VpnCreateOptions vpnLaunchOptions = VpnCreateOptions.getInstance(name, description, protocol);
+        return createVPN(vpnLaunchOptions);
```
Comment (maksimov) [inline src/main/java/org/dasein/cloud/network/AbstractVPNSupport.java]: Could be a one-liner

File: src/main/java/org/dasein/cloud/network/VPNCapabilities.java [hunk 2]
```diff
@@ -49,22 +49,6 @@
     @Nonnull Requirement getVPNVLANConstraint() throws CloudException, InternalException;
```
Comment (maksimov) [inline src/main/java/org/dasein/cloud/network/VPNCapabilities.java]: What about this one?

File: src/main/java/org/dasein/cloud/network/VPNSupport.java [hunk 3]
```diff
@@ -66,7 +66,7 @@
     @Deprecated
     public @Nonnull VPN createVPN(@Nullable String inProviderDataCenterId, @Nonnull String name, @Nonnull String description, @Nonnull VPNProtocol protocol) throws CloudException, InternalException;
     
-    public @Nonnull VPN createVPN(@Nonnull VpnLaunchOptions vpnLaunchOptions) throws CloudException, InternalException;
+    public @Nonnull VPN createVPN(@Nonnull VpnCreateOptions vpnLaunchOptions) throws CloudException, InternalException;
```
Comment (maksimov) [inline src/main/java/org/dasein/cloud/network/VPNSupport.java]: Could you clean up the 'public' modifier from this interface methods while you're at it, please?

File: src/main/java/org/dasein/cloud/network/VPNCapabilities.java (hunk 2 above)
Comment (unwin) [inline src/main/java/org/dasein/cloud/network/VPNCapabilities.java]: GCE vpn's are tied to vlan's. i think it should stay, as it determines if the .withProviderVlanId("vpn2-network") method is needed.

File: src/main/java/org/dasein/cloud/network/AbstractVPNSupport.java (hunk 1 above)
Comment (unwin) [inline src/main/java/org/dasein/cloud/network/AbstractVPNSupport.java]: wish granted!

File: src/main/java/org/dasein/cloud/network/VPNSupport.java (hunk 3 above)
Comment (unwin) [inline src/main/java/org/dasein/cloud/network/VPNSupport.java]: cleansed

File: src/main/java/org/dasein/cloud/network/AbstractVPNSupport.java [hunk 4]
```diff
@@ -26,21 +26,18 @@ public void connectToGateway(@Nonnull String providerVpnId, @Nonnull String toGa
...
-    public @Nonnull VPN createVPN(@Nonnull VpnLaunchOptions vpnLaunchOptions) throws CloudException, InternalException {
+    public @Nonnull VPN createVPN(@Nonnull VPNCreateOptions vpnLaunchOptions) throws CloudException, InternalException {
         throw new OperationNotSupportedException("createVPN is not currently implemented in " + getProvider().getCloudName());
     }
 
-    public @Nonnull VPNGateway createVPNGateway(@Nonnull String endpoint, @Nonnull String name, @Nonnull String description, @Nonnull VPNProtocol protocol, @Nonnull String bgpAsn) throws CloudException, InternalException {
-        throw new OperationNotSupportedException("createVPNGateway is not currently implemented in " + getProvider().getCloudName());
+    @Deprecated
+    public final @Nonnull VPNGateway createVPNGateway(@Nonnull String endpoint, @Nonnull String name, @Nonnull String description, @Nonnull VPNProtocol protocol, @Nonnull String bgpAsn) throws CloudException, InternalException {
+        return createVPNGateway(VPNGatewayCreateOptions.getInstance(name, description, protocol, endpoint).withBgpAsn(bgpAsn));
```
Comment (maksimov) [inline src/main/java/org/dasein/cloud/network/AbstractVPNSupport.java]: `bgpAsn` used to be required, are you sure you want to make it optional?

File: src/main/java/org/dasein/cloud/network/VPNCapabilities.java (hunk 2 above)
Comment (maksimov) [inline src/main/java/org/dasein/cloud/network/VPNCapabilities.java]: Can we at least rename it to `identifyVlanRequirement()`?

**Pull request merged**